modules/sqlite/integration/load.php

```diff
@@ -0,0 +1,150 @@
+<?php
+/**
+ * Module Name: SQLite Integration
+ * Description: Use an SQLite database instead of MySQL. <div style="background:#fff;border:1px solid #c3c4c7;border-left-width: 4px;border-left-color:#dba617;box-shadow: 0 1px 1px rgba(0, 0, 0, 0.04);padding:1em;max-width:50em;">CAUTION: Enabling this module will bring up the WordPress installation screen. You will need to reconfigure your site, and you will lose all your data. If you then disable the module, you will get back to your previous MySQL database, with all your previous data intact.</div>
+ * Experimental: Yes
+ *
+ * @package performance-lab
+ * @since x.x.x
+ */
+
+// Temporary - This will be in wp-config.php once SQLite is merged in Core.
+if ( ! defined( 'DATABASE_TYPE' ) ) {
+	define( 'DATABASE_TYPE', 'sqlite' );
+}
+
+/**
+ * Add the db.php file in wp-content.
+ *
+ * When the plugin gets merged in wp-core, this is not to be ported.
+ */
+function sqlite_plugin_copy_db_file() {
+	// Bail early if the SQLite3 class does not exist.
+	if ( ! class_exists( 'SQLite3' ) ) {
+		return;
+	}
+
+	$destination = WP_CONTENT_DIR . '/db.php';
+
+	// Bail early if the file already exists.
+	if ( file_exists( $destination ) ) {
+		return;
+	}
+
+	// Init the filesystem to allow copying the file.
+	global $wp_filesystem;
+	if ( ! $wp_filesystem ) {
+		require_once ABSPATH . '/wp-admin/includes/file.php';
+		WP_Filesystem();
+	}
+
+	if ( $wp_filesystem->touch( $destination ) ) {
+
+		// Get the db.copy file contents, replace placeholders and write it to the destination.
+		$file_contents = str_replace(
+			array(
+				'{SQLITE_IMPLEMENTATION_FOLDER_PATH}',
+				'{PERFLAB_PLUGIN}',
+				'{SQLITE_MODULE}',
+				'{PERFLAB_MODULES_SETTING}',
+			),
+			array(
+				__DIR__,
+				str_replace( WP_PLUGIN_DIR . '/', '', PERFLAB_MAIN_FILE ),
+				'sqlite/integration',
+				PERFLAB_MODULES_SETTING,
+			),
+			file_get_contents( __DIR__ . '/db.copy' )
+		);
+		$wp_filesystem->put_contents( $destination, $file_contents );
```

Comment:
As far as I understand, this is the only logic that runs when activating the module, correct? And then it would mean the user would land on a blank WP install?

Based on previous conversations we had on Slack, I would suggest that we also set up the database right away, so that it is at least a blank WordPress install with the Performance Lab plugin and the same modules enabled. This way it would make for a much better and less confusing user experience.

Would it be possible to call `wp_install()` here for the new SQLite database (at least if that database is empty)? If it's not possible within the initial request where a regular MySQL / MariaDB database is still used, we could also redirect to a custom endpoint that does this in a new request and then redirects the user back to the Performance Lab settings screen.

While this is initially going to be an experimental module and we're not going to get into the weeds of e.g. migrating a full DB, I still think a smoother user experience for this would be crucial, and the above suggestion may be feasible without too much effort.

Reply:
> As far as I understand, this is the only logic that runs when activating the module, correct? And then it would mean the user would land on a blank WP install?

Correct.

> Based on previous conversations we had on Slack, I would suggest that we also set up the database right away, so that it is at least a blank WordPress install with the Performance Lab plugin and the same modules enabled. This way it would make for a much better and less confusing user experience.

If an SQLite database doesn't exist (ie, this is the 1st time the user activates the module), they will land on the WP-install screen so they can enter their admin username/password, sitename etc.
The copied `db.php` file then takes care of activating the performance-lab plugin, and activate the default modules + SQLite.

> Would it be possible to call wp_install() here for the new SQLite database (at least if that database is empty)? If it's not possible within the initial request where a regular MySQL / MariaDB database is still used, we could also redirect to a custom endpoint that does this in a new request and then redirects the user back to the Performance Lab settings screen.

I just pushed a tweak that does that 👍 
However, I was unable to achieve it in PHP so it was accomplished using a `window.location.reload(true);` call. That in turn, triggers WP to redirect to the login screen if a database exists - and then after login the user lands on the perflab settings - OR, if a database doesn't exist, WP redirects the user to the install screen.

> While this is initially going to be an experimental module and we're not going to get into the weeds of e.g. migrating a full DB, I still think a smoother user experience for this would be crucial, and the above suggestion may be feasible without too much effort.

💯

Reply:
@aristath 
> If an SQLite database doesn't exist (ie, this is the 1st time the user activates the module), they will land on the WP-install screen so they can enter their admin username/password, sitename etc.

Why do we need to require the user to do that though? They already provided that information when they set up the original DB, so I think it would be a much smoother experience if we didn't prompt them to the WP install screen, but rather called `wp_install()` for the new DB right away, based on the values that are in the current DB.

Related is my comment https://github.com/WordPress/performance/pull/547#discussion_r1004805059, we could right after `wp_install()` also run logic to mark the Performance Lab and the current modules as active in the new SQLite DB.

Maybe I'm missing something here that makes this infeasible though?

Reply:
In the week that has passed since this comment, I tried multiple iterations and approaches to this. Unfortunately, I was unable to make it work... I agree that it would be better UX if the database was auto-populated.
However, no matter what I tried, there are issues because it's a different database, there are timing issues, we actually need to have 2 different database connections open, and overall it's a nightmare 😰 If I'm not getting an install screen, I'm getting database errors. Switching databases on the fly is no easy task 😞 

Since the "copying" functionality will not be part of Core if and when this gets merged, I don't think that getting an install screen would actually be a blocker to get this in the plugin 🤔 
If anything, it will allow us to ensure that the actual WP installation runs smoothly when a user wants to use SQLite on their site.